### ISSUE TITLE
CI test timing out: pytorch_linux_xenial_cuda10_2_cudnn7_py3_multigpu_test

### ISSUE THREAD

**Issue by bdhirsh** [pytorch/pytorch#55389]
Over the last few days, `pytorch_linux_xenial_cuda10_2_cudnn7_py3_multigpu_test` has been timing out pretty consistently on master - see https://ezyang.github.io/pytorch-ci-hud/build/pytorch-master.

It's timed out 17 out of the last 50 runs, according to the above page.

An example failure can be found [here](https://app.circleci.com/pipelines/github/pytorch/pytorch/296698/workflows/1672fac5-0b09-471b-96cd-85a323971948/jobs/12150038). The logs look something like:

```
<start>
Apr 06 14:07:13 + export 'SCRIBE_GRAPHQL_ACCESS_TOKEN=********************************************'
Apr 06 14:07:13 + SCRIBE_GRAPHQL_ACCESS_TOKEN='********************************************'

.... (lots of tests running)

Apr 06 14:15:33   test_destroy_group (__main__.TestMPIWithFork) ... skip (0.001s)
Apr 06 14:15:33   test_dump_DDP_relevant_env_vars (__main__.TestMPIWithFork) ... ok (0.002s)

Too long with no output (exceeded 1h30m0s): context deadline exceeded
<end>
```

The actual terminal output start and end timestamps only elapse around ~8 minutes of time, but the timeout implies that stuff was running for 90 minutes. Judging from that, it seems like a test that runs directly after `test_dump_DDP_relevant_env_vars` times out / deadlocks for 90 minutes, providing zero logging to the terminal.

Aside: It might be useful to log test start times in addition to end times to stdout to make future diagnosing easier?

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @seemethere @malfet @walterddr @pytorch/pytorch-dev-infra @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @agolynski @SciPioneer @H-Huang @mrzzd @cbalioglu @mruberry

**Comment by mrshenli** [comment]
@SciPioneer is working on this. This is same as https://github.com/pytorch/pytorch/issues/53899

**Comment by mrshenli** [comment]
closed by #55921
